tiatoolbox/utils/image.py

```diff
@@ -20,17 +20,183 @@
 
 """Miscellaneous utilities which operate on image data."""
 import warnings
+from typing import Tuple, Union
 
 import numpy as np
 import cv2
 from PIL import Image
 
-from tiatoolbox.utils.transforms import bounds2locsize
-from tiatoolbox.utils.misc import conv_out_size, assert_dtype_int
+from tiatoolbox.utils.transforms import bounds2locsize, imresize
+from tiatoolbox.utils.misc import conv_out_size
+
+
+PADDING_TO_BOUNDS = np.array([-1, -1, 1, 1])
+"""
+Constant array which when multiplied with padding and added to bounds,
+applies the padding to the bounds.
+"""
+# Make this immutable / non-writable
+PADDING_TO_BOUNDS.flags.writeable = False
+
+
+def normalise_padding_size(padding):
+    """Normalises padding to be length 4 (left, top, right, bottom).
+
+    Given a scalar value, this is assumed to apply to all sides and
+    therefore repeated for each output (left, right, top, bottom). A
+    length 2 input is assumed to apply the same padding to the
+    left/right and top/bottom.
+
+    Args:
+        padding (int or tuple(int)): Padding to normalise.
+
+    Raises:
+        ValueError: Invalid input size of padding (e.g. length 3).
+        ValueError: Invalid input shape of padding (e.g. 3 dimensional).
+
+    Returns:
+        :class:`numpy.ndarray`: Numpy array of length 4 with elements containing
+            padding for left, top, right, bottom.
+    """
```

Comment:
Please add a new line at the end.

Reply:
Fixed in this commit https://github.com/TIA-Lab/tiatoolbox/pull/61/commits/5b0e0ecc1fcb9ce989d907483bef9a2b0b2c7aa5